secure-api-gateway-ob-uk-rs-resource-store/secure-api-gateway-ob-uk-rs-resource-store-api/src/main/java/com/forgerock/sapi/gateway/rs/resource/store/api/admin/events/DataEventsApi.java

```diff
@@ -0,0 +1,88 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.rs.resource.store.api.admin.events;
+
+
+import java.util.Collection;
+
+import javax.validation.Valid;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import com.forgerock.sapi.gateway.ob.uk.common.error.OBErrorResponseException;
+import com.forgerock.sapi.gateway.rs.resource.store.datamodel.events.FREventMessages;
+
+import io.swagger.annotations.ApiParam;
+
+
+@RequestMapping("/admin/data")
+public interface DataEventsApi {
+
+    @RequestMapping(value = "/events",
+            produces = {"application/json; charset=utf-8"},
+            consumes = {"application/json; charset=utf-8"},
+            method = RequestMethod.POST)
+    ResponseEntity<FREventMessages> importEvents(
+            @ApiParam(value = "Default", required = true)
+            @Valid
+            @RequestBody FREventMessages frEventMessages
+    ) throws OBErrorResponseException;
+
+    @RequestMapping(value = "/events",
+            produces = {"application/json; charset=utf-8"},
+            consumes = {"application/json; charset=utf-8"},
+            method = RequestMethod.PUT)
+    ResponseEntity<FREventMessages> updateEvents(
+            @ApiParam(value = "Default", required = true)
+            @Valid
+            @RequestBody FREventMessages frEventMessages
+    ) throws OBErrorResponseException;
+
+    @RequestMapping(value = "/events/all",
+            produces = {"application/json; charset=utf-8"},
+            method = RequestMethod.GET
+    )
+    ResponseEntity<Collection<FREventMessages>> exportEvents();
+
+    @RequestMapping(value = "/events",
+            produces = {"application/json; charset=utf-8"},
+            method = RequestMethod.GET
+    )
+    ResponseEntity<FREventMessages> exportEventsByTppId(
+            @ApiParam(value = "Default", required = true)
+            @Valid
+            @RequestParam String tppId
+    );
+
+    @RequestMapping(value = "/events",
+            method = RequestMethod.DELETE
+    )
+    ResponseEntity<Void> removeEvents(
+            @ApiParam(value = "The client ID")
+            @RequestParam(value = "tppId") String tppId,
+            @ApiParam(value = "Unique identification of event message")
+            @RequestParam(value = "jti", required = false) String jti
+    );
+
+    @RequestMapping(value = "/events/all",
+            method = RequestMethod.DELETE
+    )
+    ResponseEntity<Void> removeAllEvents();
```

Comment:
Who can call this method? Seems that anyone can delete all the events?

Seems to me that this controller has a mixture of admin features (which should only be run by an ASPSP admin) and features for the TPP to manage their own events in the sandbox. I would expect that a TPP can only POST/PUT/GET/DELETE their own data.

Reply:
Deleted admin features, now the operations are restricted to the Api client that owns the events.